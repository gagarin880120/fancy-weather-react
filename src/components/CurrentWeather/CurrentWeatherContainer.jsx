import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CurrentWeather from './CurrentWeather';
import { selectCurrentWeather } from '../../redux/selectors';

export function CurrentWeatherContainer({ currentWeather }) {
  // console.log(currentWeather);
  return (
    currentWeather
      ? (
        <CurrentWeather
          currentWeather={currentWeather}
        />
      ) : <div>Loading...</div>
  );
}

export const mapStateToProps = (state) => ({
  currentWeather: selectCurrentWeather(state),
});

CurrentWeatherContainer.propTypes = {
  currentWeather: PropTypes.shape({}),
};

CurrentWeatherContainer.defaultProps = {
  currentWeather: {},
};

export default connect(mapStateToProps)(CurrentWeatherContainer);
