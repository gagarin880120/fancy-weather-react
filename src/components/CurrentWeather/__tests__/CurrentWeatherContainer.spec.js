import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CurrentWeatherContainer, mapStateToProps } from '../CurrentWeatherContainer';

describe('CurrentWeatherContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<CurrentWeatherContainer />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    test('should return right value', () => {
      const initialState = {
        language: 'en',
        temperatureScale: 'celsius',
        currentWeather: {
          rh: 15,
          temp: 12,
          wind_spd: 3,
          weather: {
            code: '801',
            icon: 'c02d'
          }
        },
      }

      expect(mapStateToProps(initialState)).toEqual({
        currentWeather: {
          rh: 'Humidity: 15%',
          wind_spd: 'Wind: 3m/s',
          temp: 12,
          icon: 'DayCloudyGusts',
          weather: {
            code: '801',
            icon: 'c02d',
          },
        },
      });
    });
  });
});
