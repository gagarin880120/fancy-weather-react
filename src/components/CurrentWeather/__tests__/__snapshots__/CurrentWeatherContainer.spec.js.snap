// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeatherContainer component should render without crashing 1`] = `
<CurrentWeather
  currentWeather={Object {}}
/>
`;
